File: c++/RearrangeArrayAccordingToIndex.cpp
#include <iostream>
using namespace std;

void answer(int arr[], int n){
	int i, j, temp;

	for (i = 0;i<n;i++){
		for (j = 0;j<n;j++){
			if (arr[j] == i) {
				temp = arr[j];
				arr[j] = arr[i];
				arr[i] = temp;
				break;
			}
		}
	}

	for (i = 0;i<n;i++){
		if (arr[i] != i)
			arr[i] = -1;		
	}

	for (i = 0;i<n;i++) {
		cout<<arr[i];
	}
}

int main(){
	int n, arr[] = { -8, 1, 3, 7, 10, 12, 2, 2001, 2002, -666};
	n = sizeof(arr) / sizeof(arr[0]);

	answer(arr, n);
}

File: python/RearrangeArrayAccordingToIndex.py
def answer(arr, n):
	
	for i in range(n):
		for j in range(n):
			if (arr[j] == i):
				arr[j], arr[i] = arr[i], arr[j]


	for i in range(n):		
		if (arr[i] != i):
			arr[i] = -1


	for i in range(n):
		print(arr[i])


arr = [-8, 1, 3, 7, 10, 12, 2, 2001, 2002, -666]

n = len(arr)

answer(arr, n);
